### PR TITLE
Ex 05

### PR DESCRIPTION
**Pull request by nimve**
Here is my submission for Exercise 5. Looking forward to any comments!

### CODE REVIEW

File: src/nina_vesseltun_ex/ex05/bounded_sim.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self, start, home, left_lim, right_lim):
+        super().__init__(start, home)
+        self.left_limit = left_lim
+        self.right_limit = right_lim
+
+    def move(self):
+        if randint(0, 1):
+            if self.x < self.right_limit:
+                self.x += 1
+                self.steps += 1
```
Comment (huynhngoc) [inline src/nina_vesseltun_ex/ex05/bounded_sim.py]: Even if it cannot move, it still counts as a step. So put the increment for steps outside the if block

File: src/nina_vesseltun_ex/ex05/bounded_sim.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.right_limit = right_lim
+
+    def single_walk(self):
+        pedestrian = BoundedWalker(self.start, self.home, self.left_limit,
+                                   self.right_limit)
+        play = True
+        while play:
+            pedestrian.move()
+            if pedestrian.is_at_home():
+                play = False
```
Comment (huynhngoc) [inline src/nina_vesseltun_ex/ex05/bounded_sim.py]: remove these 2 lines

File: src/nina_vesseltun_ex/ex05/bounded_sim.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    def __init__(self, start, home, seed, left_lim, right_lim):
+        super().__init__(start, home, seed)
+        self.left_limit = left_lim
+        self.right_limit = right_lim
+
+    def single_walk(self):
+        pedestrian = BoundedWalker(self.start, self.home, self.left_limit,
+                                   self.right_limit)
+        play = True
+        while play:
```
Comment (huynhngoc) [inline src/nina_vesseltun_ex/ex05/bounded_sim.py]: Then change this to
`while not pedestrian.is_at_home():`

File: src/nina_vesseltun_ex/ex05/walker_sim.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        self.seed = the_seed
+
+    def single_walk(self):
+        pedestrian = Walker(self.start, self.home)
+        play = True
+        while play:
+            pedestrian.move()
+            if pedestrian.is_at_home():
+                play = False
+        return pedestrian.get_steps()
```
Comment (huynhngoc) [inline src/nina_vesseltun_ex/ex05/walker_sim.py]: See bounded_sim

File: src/nina_vesseltun_ex/ex05/myrand.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            raise RuntimeError(
+                'Can only be initialised as an iterator once'
+            )
+        self.num_generated_numbers = 0
+        return self
+
+    def __next__(self):
+        if self.num_generated_numbers is None:
+            raise RuntimeError(
+                'Cannot call ``next`` before the Polygon is initialised'
```
Comment (huynhngoc) [inline src/nina_vesseltun_ex/ex05/myrand.py]: Are you copying code from somewhere?

**Review by huynhngoc — COMMENTED** [review]
Some early feedback for you to change.

**Comment by nimve** [comment]
I've made the changes, please have another look when you find the time. As for the polygon-code, it was left there by mistake after experimenting with notes from the lecture at an early stage to understand the concept of __iter__ and __next__, and never meant to be included in the final submission. Thanks for pointing it out!

File: src/nina_vesseltun_ex/ex05/myrand.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+class LCGRand:
+    a = 16807
+    m = 2 ** 31 - 1
+
+    def __init__(self, seed):
+        self.seed = seed
+
+    def rand(self):
+        while True:
```
Comment (huynhngoc) [inline src/nina_vesseltun_ex/ex05/myrand.py:15]: remove this line.

File: src/nina_vesseltun_ex/ex05/walker_sim.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        self.seed = the_seed
+
+    def single_walk(self):
+        pedestrian = Walker(self.start, self.home)
+        while not pedestrian.is_at_home():
+            pedestrian.move()
+        return pedestrian.get_steps()
+
+    def run_simulation(self, num_walks):
+        seed(self.seed)
```
Comment (huynhngoc) [inline src/nina_vesseltun_ex/ex05/walker_sim.py:47]: You should put this in `__init__` as random seed should be set once in the very first start.

**Review by huynhngoc — APPROVED** [review]
Nice work. Approved.

**Pull request merged**